ext/pdo_mysql/mysql_sql_parser.re

```diff
@@ -0,0 +1,48 @@
+/*
+  +----------------------------------------------------------------------+
+  | Copyright (c) The PHP Group                                          |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 3.01 of the PHP license,      |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | https://www.php.net/license/3_01.txt                                 |
+  | If you did not receive a copy of the PHP license and are unable to   |
+  | obtain it through the world-wide-web, please send a note to          |
+  | license@php.net so we can mail you a copy immediately.               |
+  +----------------------------------------------------------------------+
+  | Author: Matteo Beccati <mbeccati@php.net>                            |
+  +----------------------------------------------------------------------+
+*/
+
+
+#include "php.h"
+#include "pdo/php_pdo_driver.h"
+#include "pdo/php_pdo_int.h"
+#include "pdo/pdo_sql_parser.h"
+
+int pdo_mysql_scanner(pdo_scanner_t *s)
+{
+	const char *cursor = s->cur;
+
+	s->tok = cursor;
+	/*!re2c
+	BINDCHR		= [:][a-zA-Z0-9_]+;
+	QUESTION	= [?];
+	COMMENTS	= ("/*"([^*]+|[*]+[^/*])*[*]*"*/"|("--"|[#])[^\r\n]*);
+	SPECIALS	= [:?"'`/#-];
+	MULTICHAR	= ([:]{2,}|[?]{2,});
+	ANYNOEOF	= [\001-\377];
+	*/
+
+	/*!re2c
+		(["]((["]["])|([\\]ANYNOEOF)|ANYNOEOF\["\\])*["]) { RET(PDO_PARSER_TEXT); }
```

Comment:
this should be configureable somehow based on the used sql_mode - https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sqlmode_no_backslash_escapes

Reply:
Sorry, that's out of scope for the current RFC. Multiple parsers for each SQL mode is a bit overkill. I'd be happy to document what are the suggested settings.